src/wallet/wallet.ts

```diff
@@ -2154,6 +2465,313 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
+    const addressIndex = addressDetails?.index;
+
+    if (addressIndex === undefined) {
+      throw new Error(
+        `Address used to sign the transaction (${address}) does not belong to the wallet.`
+      );
+    }
+
+    const storageProxy = new WalletServiceStorageProxy(this, this.storage);
+
+    await transaction.signTransaction(tx, storageProxy.createProxy(), pinCode);
+
+    // Finalize the transaction
+    tx.prepareToSend();
+
+    const sendTransaction = new SendTransactionWalletService(this, {
+      transaction: tx,
+      pin: pinCode,
+    });
+    return sendTransaction;
+  }
+
+  /**
+   * Create a Create Token Transaction with nano header and return the SendTransaction object
+   *
+   * @param {string} method Method of nano contract to have the transaction created
+   * @param {string} address Address that will be used to sign the nano contract transaction
+   * @param {object} data Nano contract data (blueprintId, ncId, actions, args)
+   * @param {object} createTokenOptions Options for token creation (mint/melt authorities, NFT, etc)
+   * @param {object} options Options (pinCode)
+   *
+   * @returns {Promise<SendTransactionWalletService>}
+   */
+  async createNanoContractCreateTokenTransaction(
+    method: string,
+    address: string,
+    data: CreateNanoTxData,
+    createTokenOptions: CreateTokenOptionsInput,
+    options: { pinCode?: string } = {}
+  ): Promise<SendTransactionWalletService> {
+    this.failIfWalletNotReady();
+    if (await this.storage.isReadonly()) {
+      throw new WalletFromXPubGuard('createNanoContractCreateTokenTransaction');
+    }
+    const newOptions = { pinCode: null, ...options };
+    const pin = newOptions.pinCode;
+    if (!pin) {
+      throw new PinRequiredError('Pin is required.');
+    }
+    // Validate address belongs to wallet and get its index
+    const addressDetails = await this.getAddressDetails(address);
+    const addressIndex = addressDetails?.index;
+    if (addressIndex === undefined) {
+      throw new Error(
+        `Address used to sign the transaction (${address}) does not belong to the wallet.`
+      );
+    }
+    // Get the address public key
+    const addressPrivKey = await this.getAddressPrivKey(pin, addressIndex);
+    const callerAddress = new Address(
+      addressPrivKey.publicKey.toAddress(this.network.getNetwork()).toString(),
+      { network: this.network }
+    );
+
+    // Build and send transaction
+    const actions = data.actions || [];
+    const args = data.args || [];
+    // Process arguments to handle serialized Buffer objects
+    const processedArgs = args.map(arg => {
+      if (
+        arg &&
+        typeof arg === 'object' &&
+        arg !== null &&
+        'data' in arg &&
+        'type' in arg &&
+        (arg as { type?: string }).type === 'Buffer' &&
+        Array.isArray((arg as { data?: unknown }).data)
+      ) {
+        return Buffer.from((arg as { data: number[] }).data);
+      }
+      return arg;
+    });
```

Comment:
This part of the code seems the same as in the `createNanoContractTransaction` method. Can we reuse it somehow?

Reply:
Yes, done